### ISSUE TITLE
Add option for filtering commits by path

### ISSUE THREAD

**Issue by iloire** [lob/generate-changelog#13]
I have three components in the same repository. It would be awesome if I could  filter them by path (``git log -- <path>``), so I can generate three different CHANGELOG files for each component.

Any plans to add this? Would you accept a PR?

**Comment by robinjoseph08** [comment]
Yeah, that's a good idea! We'd absolutely be willing to accept PRs.

**Comment by altJake** [comment]
I think I feel comfortable enough to encounter that, and will give it a try.
One question though, what should we do if there is a conflict between this new `subpath` option and the output file option?
e.g. if there is also an output file specified in a different than the current folder. I thought to address that this way: 
 - If there is an explicit output file specified in a **different path than the current directory** --> I will use it as is
 - If the output file left defaulted, or is designated to be in the current directory **and** there is `subpath` option set --> I will place/work on a file in that provided subpath

@iloire @robinjoseph08 Thoughts?

**Comment by altJake** [comment]
@iloire @robinjoseph08 

Feel free to give a look / try / test / comment / suggest

**Comment by solarmosaic-kflorence** [comment]
This tool is unusable in a monorepo without this feature.
